internal/fips/fipscheck.go

```diff
@@ -0,0 +1,88 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fips
+
+import (
+	"strings"
+)
+
+type FIPSCheck interface {
+	DisabledComponents(receivers map[string]interface{}, exporters map[string]interface{}, processors map[string]interface{}, extensions map[string]interface{}) []string
+}
+
+// FipsCheck holds configuration for FIPS black list.
+type fipsCheck struct {
+	receivers  map[string]bool
+	exporters  map[string]bool
+	processors map[string]bool
+	extensions map[string]bool
+}
+
+type noopFIPSCheck struct{}
+
+func (noopFIPSCheck) DisabledComponents(receivers map[string]interface{}, exporters map[string]interface{}, processors map[string]interface{}, extensions map[string]interface{}) []string {
+	return nil
+}
+
+// NewFipsCheck creates new FipsCheck.
+// It checks if FIPS is enabled on the platform in /proc/sys/crypto/fips_enabled.
+func NewFipsCheck(FIPSEnabled bool, receivers, exporters, processors, extensions []string) FIPSCheck {
```

Comment:
at the call site, we should just create the FipsCheck if its enabled, otherwise we set the interface to be nil i.e.

```
var fipsCheck fips.FIPSCheck
if ad.FIPSEnabled(ctx) {
    fips.NewFipsCheck(receivers, exporters, processors, extensions)
}
...
```

Reply:
I prefer to avoid dealing with `nil` checks in the webhook or wherever the check is used.

Reply:
that's how we handle other conditional checks that the webhook is supposed to be running today ([here](https://github.com/open-telemetry/opentelemetry-operator/blob/main/apis/v1beta1/collector_webhook.go#L48-L50)) so it feels odd to special case this. I suppose if we want to avoid that we could just create the NoopCheck here instead. I don't love the bool in the signature, it feels like a leak in terms of responsibility for the package.